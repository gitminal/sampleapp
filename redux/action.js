export const PRODUCT_SHOW = 'PRODUCT_SHOW';
export const GET_APISHOW = 'GET_APISHOW';
export const ERROR_SHOW = 'ERROR_SHOW';
import axios from "axios";

export const getproduct = () => {
  try {
    return async dispatch => {
      const res = await axios.get("https://makeup-api.herokuapp.com/api/v1/products.json?brand=maybelline");
      console.log("res", res.data[0].name);
      if (res.data) {

        dispatch({
          type: PRODUCT_SHOW,
          payload: res.data,

        })
      } else {
        dispatch({ type: ERROR_SHOW, payload: "unable to fetch" });

      }
    };
  } catch (error) {
    dispatch({ type: ERROR_SHOW, payload: error });
  }
};