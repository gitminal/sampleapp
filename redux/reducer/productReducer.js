import {ERROR_SHOW} from '../action/utility';
import {GET_APISHOW} from '../action/utility';
import {PRODUCT_SHOW} from '../action/utility';

 export const productstate={
product:[],
error: '',
loading: false
};

function productReducer(state = productstate, action) {
  switch (action.type) {
     case GET_APISHOW:
        return {...state,loading:true};
    case PRODUCT_SHOW:
      return {...state, product: action.payload,loading:false};
    case ERROR_SHOW:
      return {...state,error:action.payload,loading:false};
      default:
      return state;
  }
}
export default productReducer;