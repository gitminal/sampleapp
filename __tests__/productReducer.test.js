import {PRODUCT_SHOW} from '../redux/action/utility';
import { productstate } from "../redux/reducer/productReducer";
import product from "../redux/reducer/productReducer"
describe('the user reducer', () => {
  const productstate = {
    product:[],
    error: '',
    loading: false
  };

  it('should return the initialState', () => {
    expect(product(undefined, {})).toEqual(productstate);
  });

  it('should handle the GET_APISHOW action', () => {
    const action = { type: 'GET_APISHOW' };
    const expectedState = { ...productstate, loading: true };

    expect(product(productstate, action)).toEqual(expectedState);
  });

  it('should handle the PRODUCT_SHOW action', () => {
    const productlist = [{name:"liner",price:"20"}];
    const action = {
      type: 'PRODUCT_SHOW',
      payload: { product: productlist },
    };
    const expectedState = { ...productstate, product: productlist };

    expect(product(productstate, action)).toMatchSnapshot();
  });
  
  it('should handle the ERROR_SHOW action', () => {
    const action = { type: 'ERROR_SHOW' };
    const expectedState = { ...productstate, loading: false };

    expect(product(productstate, action)).toMatchSnapshot();
  });
});




