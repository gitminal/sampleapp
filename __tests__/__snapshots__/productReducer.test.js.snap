// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the user reducer should handle the ERROR_SHOW action 1`] = `
Object {
  "error": undefined,
  "loading": false,
  "product": Array [],
}
`;

exports[`the user reducer should handle the PRODUCT_SHOW action 1`] = `
Object {
  "error": "",
  "loading": false,
  "product": Object {
    "product": Array [
      Object {
        "name": "liner",
        "price": "20",
      },
    ],
  },
}
`;
